manager/integration/tests/test_migration.py

```diff
@@ -147,7 +147,40 @@ def test_migration_with_failed_replica(clients, volume_name):  # NOQA
        And the old failed replica will be cleaned up.
     8. Validate initially written test data.
     """
-    return
+    client, volume, data = setup_migration_test(clients,
+                                                volume_name,
+                                                replica_cnt=3)
+
+    current_node = common.get_self_host_id()
+    host1, host2 = get_hosts_for_migration_test(clients)
+    volume.attach(hostId=current_node)
+    volume = common.wait_for_volume_healthy(client, volume_name)
+
+    exec_cmd = ["rm", "-rf",  "/var/lib/longhorn/replicas"]
+    subprocess.check_output(exec_cmd)
+    volume = common.wait_for_volume_degraded(client, volume_name)
+
+    volume.attach(hostId=host1)
+
+    volume = common.wait_for_volume_migration_ready(client, volume_name)
+
+    volume.detach(hostId=current_node)
+
+    volume = common.wait_for_volume_migration_node(client,
+                                                   volume_name,
+                                                   host1)
+
+    volume.updateReplicaCount(replicaCount=2)
+    volume.detach(hostId="")
+    volume = common.wait_for_volume_detached(client, volume_name)
+    volume = client.by_id_volume(volume_name)
```

Comment:
Why does the test need this? And how can it verify the data if the volume is detached?

Reply:
Hi @shuo-wu, `check_volume_data` will attach the volume and doing data check

Reply:
Modified function name from `check_volume_data` to `check_detached_volume_data`